turbopack/crates/turbopack-ecmascript-runtime/js/src/browser/runtime/base/dev-base.ts

```diff
@@ -257,22 +248,30 @@ function runModuleExecutionHooks(
   module: Module,
   executeModule: (ctx: RefreshContext) => void
 ) {
-  const cleanupReactRefreshIntercept =
-    typeof globalThis.$RefreshInterceptModuleExecution$ === 'function'
-      ? globalThis.$RefreshInterceptModuleExecution$(module.id)
-      : () => {}
-
-  try {
+  if (typeof globalThis.$RefreshInterceptModuleExecution$ === 'function') {
+    const cleanupReactRefreshIntercept =
+      globalThis.$RefreshInterceptModuleExecution$(module.id)
+    try {
+      executeModule({
+        register: globalThis.$RefreshReg$,
+        signature: globalThis.$RefreshSig$,
+        registerExports: registerExportsAndSetupBoundaryForReactRefresh,
+      })
+    } finally {
+      // Always cleanup the intercept, even if module execution failed.
+      cleanupReactRefreshIntercept()
+    }
+  } else {
+    // In a web worker, we don't install the react refresh hooks so bind dummy functions.
+    if (typeof importScripts !== 'function') {
+      // But if we aren't in a web worker, emit a warning.  Something is wrong.
+      console.warn(`React refresh support is not enabled.  HMR will not work.`)
```

Comment:
This warning is causing a bunch of tests to fail...

im guessing it is quite intentional that HMR is disabled? should i just drop this or is there some better signal that HMR is intentionally not supported in a dev environment?

Reply:
Showing this message for every module is also a bit verbose...

Reply:
Ok, ill just drop it.  @wbinnssmith was concerned about this approach hiding misconfigurations, but module instantiation probably isn't the best place to flag it anyway